src/web/extractors.rs

```diff
@@ -1269,6 +1291,43 @@ impl FromRequest for BsoQueryParams {
                 Some(tags.clone()),
             )
         })?;
+        if params.sort != Sorting::Index {
+            let bound = match params
+                .offset
+                .clone()
+                .unwrap_or(Offset {
+                    timestamp: None,
+                    offset: 0,
+                })
+                .timestamp
+            {
+                None => 0,
+                Some(timestamp) => timestamp.as_i64(),
+            };
```

Comment:
Not really sure you need the match here. You could have the `unwrap_or` set the `timestamp` to 0 and then convert `.timestamp.as_i64()`